src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -281,6 +285,91 @@ int get(byte[] key, ByteBuffer buffer) throws IOException {
         }
     }
 
+    public boolean takeSnapShot() {
+        synchronized (snapshotLock) {
+            logger.info("Start generating the snapshot");
+
+            try {
```

Comment:
Before start, check ```isClosing```, ```isOpening``` (to be added) and ```isTombstoneFilesMerging```, if any one is true, should return.

Reply:
Nice points.